queries/perl/highlights.scm

```diff
@@ -1,198 +1,153 @@
-  (special_array_variable)
-  (special_hash_variable)
-  (special_literal)
-  (super)
-] @variable.builtin
+(_ operator: _ @operator)
+"\\" @operator
 
-((scalar_variable) @variable.builtin
- (#eq? @variable.builtin "$#ARGV"))
+(yadayada) @exception
 
-; Integer numbers
-[
-  (integer)
-  (hexadecimal)
-] @number
+(phaser_statement phase: _ @keyword.phaser)
 
-; Float numbers
 [
-  (floating_point)
-  (scientific_notation)
-] @float
-
-; version sortof counts as a kind of multipart integer
-(version) @constant
-
-; Package names are types
-(package_name) @type
-
-; The special SUPER:: could be called a namespace. It isn't really but it
-;   should highlight differently and we might as well do it this way
-(super) @namespace
-
-; Comments are comments
-(comments) @comment @spell
-
-((source_file . (comments) @preproc)
-  (#lua-match? @preproc "^#!/"))
-
-; POD should be handled specially with its own embedded subtype but for now
-;   we'll just have to do this.
-(pod_statement) @text
+  "or" "and"
+  "eq" "ne" "cmp" "lt" "le" "ge" "gt"
+  "isa"
+] @keyword.operator
 
-(method_invocation
-  function_name: (identifier) @method.call)
-(call_expression
-  function_name: (identifier) @function.call)
+(eof_marker) @preproc
+(data_section) @comment
 
-; Built-in functions
-((call_expression
-  function_name: (identifier) @function.builtin)
- (#any-of? @function.builtin
-   "print" "printf" "sprintf" "say"
-   "push" "pop" "shift" "unshift" "splice"
-   "exists" "delete" "keys" "values"
-   "each"))
+(pod) @text
 
 [
-  (map)
-  (grep)
-  (bless)
-] @function.builtin
+  (number)
+  (version)
+] @number
 
-;; ----------
+[
+  (string_literal) 
+  (interpolated_string_literal) 
+  (quoted_word_list) 
+  (command_string) 
+  (heredoc_content)
+] @string
 
-(use_constant_statement
-  constant: (identifier) @constant)
+[
+  (heredoc_token)
+  (command_heredoc_token)
+  (heredoc_end)
+] @label
 
-(named_block_statement
-  function_name: (identifier) @function)
+[(escape_sequence) (escaped_delimiter)] @string.escape
 
-(function_definition
-  name: (identifier) @function)
+(_ modifiers: _ @character.special)
+[  
+ (quoted_regexp)
+ (match_regexp)
+] @string.regex
 
-(function) @function
+(regexp_content) @string.regex
+(replacement) @string
 
-[
-  "(" ")"
-  "[" "]"
-  "{" "}"
-  (standard_input_to_variable)
-] @punctuation.bracket
+(transliteration_content) @string
 
-[ "`" "\"" ] @punctuation.special
+(autoquoted_bareword _?) @string.special
 
-[
-  "=~"
-  "!~"
-  "="
-  "=="
-  "+"
-  "-"
-  "."
-  "//"
-  "||"
-  "&&"
-  "<<"
-  (arrow_operator)
-  (hash_arrow_operator)
-  (array_dereference)
-  (hash_dereference)
-  (to_reference)
-  (type_glob)
-  (hash_access_variable)
-] @operator
+(use_statement (package) @type)
+(package_statement (package) @type)
+(require_expression (bareword) @type)
 
-[
-  (regex_option)
-  (regex_option_for_substitution)
-  (regex_option_for_transliteration)
-] @parameter
+(subroutine_declaration_statement name: (_) @function)
```

Comment:
can a name be different rules? e.g. in Lua it could be like `function foo()` or `function M.foo()`, but you wouldn't want to capture `M.foo`, just `foo`, so ideally capturing identifiers are what you want, but maybe this is correct, not familiar w/ Perl